curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md

```diff
@@ -42,7 +42,7 @@ assert.match(code, /\s*setTimeout\s*\(\s*\(\s*\)\s*=>[^,]*,\s*2000\s*\)/s);
 Your callback function should set the `checkpointScreen` `display` property to `none`.
 
 ```js
-assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*\(\s*\)\s*=>\s*\(\s*checkpointScreen\.style\.display\s*=\s*["']none["']\s*\)\s*,\s*2000\s*\)\s*;?\s*}/s);
+assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*\(\s*\)\s*=>\s*(\(\s*checkpointScreen\.style\.display\s*=\s*("|')none\2\s*\)|\{\s*checkpointScreen\.style\.display\s*=\s*("|')none\3\s*\}|\s*checkpointScreen\.style\.display\s*=\s*("|')none\4\s*)\s*,\s*2000\s*\)\s*;?\s*}/s);
```

Comment:
This still fails the option that was laid out in the issue 

```js
  if(isCheckpointCollisionDetectionActive) {
    setTimeout(() => { checkpointScreen.style.display = "none"; }, 2000);
  }
```
<img width="885" alt="Screenshot 2024-01-04 at 8 30 27 AM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/8d20af81-ff31-4bda-9d33-fe97dd42333a">

this should fix it though
<img width="890" alt="Screenshot 2024-01-04 at 8 34 16 AM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/8c28dbab-734f-4b10-9508-f87807176961">

```suggestion
assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*\(\s*\)\s*=>\s*(\(\s*checkpointScreen\.style\.display\s*=\s*("|')none\2\s*\)|\{\s*checkpointScreen\.style\.display\s*=\s*("|')none\3\s*;?\s*\}|\s*checkpointScreen\.style\.display\s*=\s*("|')none\4\s*)\s*,\s*2000\s*\)\s*;?\s*}/s);
```

Reply:
I though I was copying the example directly when I was testing changed regex... Ah, the optional semicolon strucking when you expect it the least! :eyes: